beacon-chain/cache/depositcache/deposits_cache.go

```diff
@@ -163,6 +163,7 @@ func (dc *DepositCache) DepositByPubkey(ctx context.Context, pubKey []byte) (*et
 
 	var deposit *ethpb.Deposit
 	var blockNum *big.Int
+	// This is rather slow, anyway we can speed it up?
```

Comment:
So this needs to be an ordered slice: https://github.com/prysmaticlabs/prysm/blob/52e6cb1046be0b9a95bd8e0b0e6c9e6431629209/beacon-chain/cache/depositcache/deposits_cache.go#L72. What we could do is we could keep a map of pubkey prefix to boolean as a way to do quick checks. If there's a cache miss, then we can iterate through this whole slice. That could be a neat approach.

Reply:
It might be better to search starting from the end also, instead of starting with all the genesis deposits, wdyt?